### PR TITLE
turso-cli: init at 0.79.0

### PR DESCRIPTION
**Pull request by zestsystem**
## Description of changes
For `turso dev` command to work the following PR for sqld should be merged: 
https://github.com/NixOS/nixpkgs/pull/246426

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/turso-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}:
+buildGoModule rec {
+  pname = "turso-cli";
+  version = "0.79.0";
+
+  src = fetchFromGitHub {
+    owner = "tursodatabase";
+    repo = "turso-cli";
+    rev = "v${version}";
+    hash = "sha256-+GU8yUdOfWbN/eKUFdYDXCVNVD+Klul1Q6PX6rDVeKw=";
```
Comment (drupol) [inline pkgs/development/tools/turso-cli/default.nix:14]:
```suggestion
    hash = "sha256-5ucStAFe3lZgnGMI0fRw1E4T60+9nglNbZnzrjRmRgk=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
Hi!

Thanks for implementing the requested changes. Could you please squash your commits so there's only 2?

Thanks

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -18714,6 +18714,12 @@
     github = "zeri42";
     githubId = 68825133;
   };
+  zestsystem = {
+     name = "Mike Yim";
+     email = "mk337337@gmail.com";
+     github = "zestsystem";
+     githubId = 39456023;
```
Comment (drupol) [inline maintainers/maintainer-list.nix:18721]: Keep these attributes sorted alphabetically.

File: pkgs/development/tools/turso-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "tursodatabase";
+    repo = "turso-cli";
+    rev = "v${version}";
+    hash = "sha256-5ucStAFe3lZgnGMI0fRw1E4T60+9nglNbZnzrjRmRgk=";
+  };
+
+  vendorHash = "sha256-+F9I6+f7Sm5qhBAoXCMKjV/jFY0fyVIk0NKBQNNI+qM=";
+
+  doCheck = false;
```
Comment (drupol) [inline pkgs/development/tools/turso-cli/default.nix:19]: Why the tests are disabled?

**Review by drupol — CHANGES_REQUESTED** [review]
Further remarks

File: pkgs/development/tools/turso-cli/default.nix (hunk 3 above)
Comment (zestsystem) [reply pkgs/development/tools/turso-cli/default.nix:19]: `Test_setDatabasesCache` fails due to `/homeless-shelter: read-only file system error`. The test uses `configdir.MakePath` to try to make `/homeless-shelter/Library/Application` `Support/turso`. If there is a workaround for this that I am missing please let me know :)

Comment (drupol) [reply pkgs/development/tools/turso-cli/default.nix:19]: Ok, can you please add a comment in the pr explaining in one sentence why they are disabled?

**Pull request merged**